var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -696,6 +706,35 @@ def post_configure_fixes(self):
         filter_file('build:opt --host_copt=-march=native', '',
                     '.tf_configure.bazelrc')
 
+        if spec.satisfies('+rocm'):
+            filter_file(r'\@SPACK_HIP_ROOT\@',
+                        '"{0}"'.format(spec['hip'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_ROCFFT_ROOT\@',
+                        '"{0}"'.format(spec['rocfft'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_ROCBLAS_ROOT\@',
+                        '"{0}"'.format(spec['rocblas'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_MIOPEN_ROOT\@',
+                        '"{0}"'.format(spec['miopen-hip'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_RCCL_ROOT\@',
+                        '"{0}"'.format(spec['rccl'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_HIPRAND_ROOT\@',
+                        '"{0}"'.format(spec['rocrand'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_HIPSPARSE_ROOT\@',
+                        '"{0}"'.format(spec['hipsparse'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_ROCPRIM_ROOT\@',
+                        '"{0}"'.format(spec['rocprim'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
+            filter_file(r'\@SPACK_HIPCUB_ROOT\@',
+                        '"{0}"'.format(spec['hipcub'].prefix),
+                        'third_party/gpus/rocm_configure.bzl')
```

Comment:
This can be simplified a bit:

```suggestion
            rocm_configure = FileFilter("third_party/gpus/rocm_configure.bzl")
            rocm_configure.filter(r'\@SPACK_HIP_ROOT\@',
                        '"{0}"'.format(spec['hip'].prefix))
            rocm_configure.filter(r'\@SPACK_ROCFFT_ROOT\@',
                        '"{0}"'.format(spec['rocfft'].prefix))
            rocm_configure.filter(r'\@SPACK_ROCBLAS_ROOT\@',
                        '"{0}"'.format(spec['rocblas'].prefix))
            rocm_configure.filter(r'\@SPACK_MIOPEN_ROOT\@',
                        '"{0}"'.format(spec['miopen-hip'].prefix))
            rocm_configure.filter(r'\@SPACK_RCCL_ROOT\@',
                        '"{0}"'.format(spec['rccl'].prefix))
            rocm_configure.filter(r'\@SPACK_HIPRAND_ROOT\@',
                        '"{0}"'.format(spec['rocrand'].prefix))
            rocm_configure.filter(r'\@SPACK_HIPSPARSE_ROOT\@',
                        '"{0}"'.format(spec['hipsparse'].prefix))
            rocm_configure.filter(r'\@SPACK_ROCPRIM_ROOT\@',
                        '"{0}"'.format(spec['rocprim'].prefix))
            rocm_configure.filter(r'\@SPACK_HIPCUB_ROOT\@',
                        '"{0}"'.format(spec['hipcub'].prefix))
```

Reply:
i have addressed the changes and updated the branch with the latest develop. Will try to resolve the build errors when building the bazel package. it is trying to access {rocm-path}/.info/version-dev file which we do not support currently.